metasynth/distribution/categorical.py

```diff
@@ -42,33 +42,52 @@ def __init__(self, labels: Union[npt.NDArray[np.str_], list[str]],
     def _fit(cls, values: pl.Series):
         labels, counts = np.unique(values, return_counts=True)
         probs = counts/np.sum(counts)
-        return cls(labels.astype(str), probs)
+        return cls(labels, probs)
 
     def _param_dict(self):
         return {"labels": self.labels, "probs": self.probs}
 
     @classmethod
     def _param_schema(cls):
         return {
-            "labels": {"type": "array", "items": {"type": "string"}, "uniqueItems": True},
+            "labels": {"type": "array", "uniqueItems": True},
             "probs": {"type": "array", "items": {"type": "number"}},
         }
 
     def draw(self):
-        return str(np.random.choice(self.labels, p=self.probs))
+        return np.random.choice(self.labels, p=self.probs)
 
     def information_criterion(self,
                               values: Union[pd.Series, pl.Series, npt.NDArray[np.str_]]
                               ) -> float:
-        values_array = np.array(values, dtype=str)
-        labels, counts = np.unique(values_array, return_counts=True)
+        series = self._to_series(values)
+        labels, counts = np.unique(series, return_counts=True)
         log_lik = 0.0
         pdict = dict(zip(self.labels, self.probs))
+        if len(self.labels) > 1 and isinstance(self.labels[0], (int, np.int_)):
+            # Do add-one smoothing, assume count of values is equal to one in initial fit.
```

Comment:
Why add-one smoothing and not add-half smoothing?

Reply:
This comment should also be more explicit, why are we doing this, what effects does it have? or documented better in the docstring.

Reply:
> Why add-one smoothing and not add-half smoothing?

I don't really have any solid grounds on it, except it being argued in the Wikipedia page. Again, since I think the multinoulli distribution is more of a "last resort" type of distribution, if 1 is reasonable it might be preferable? If you think 1/2 is better defensible than 1, then we can put it to 1/2, but from what I read it seems like it depends on your assumption/taste.